src/api/usage.ts

```diff
@@ -0,0 +1,20 @@
+import { Router } from "./router";
+import { UsageResponse } from "../types";
+
+/**
+ * Usage API Interface:
+ * Get information about API usage and credits.
+ * https://docs.dune.com/api-reference/usage/endpoint/get-usage
+ */
+export class UsageAPI extends Router {
+  /**
+   * Get usage information for the current API key.
+   * Returns details about private queries, dashboards, storage,
+   * and credit usage for billing periods.
+   * @returns {Promise<UsageResponse>} usage information including credits and storage
+   */
+  async getUsage(): Promise<UsageResponse> {
+    const response = await this.post<UsageResponse>("usage");
+    return response;
```

Comment:
I don't think we need to declare a const or await here.

```suggestion
  async getUsage(): Promise<UsageResponse> {
    return this.post<UsageResponse>("usage");
```